pkg/kubelet/cm/topologymanager/policy_single_numa_node.go

```diff
@@ -52,8 +54,122 @@ func (p *singleNumaNodePolicy) canAdmitPodResult(hint *TopologyHint) lifecycle.P
 	}
 }
 
+// Merge a TopologyHints permutation to a single hint by performing a bitwise-AND
+// of their affinity masks. At this point, all hints have single numa node affinity
+// and preferred-true.
+func (p *singleNumaNodePolicy) mergePermutation(permutation []TopologyHint) TopologyHint {
+	preferred := true
+	var numaAffinities []bitmask.BitMask
+	for _, hint := range permutation {
+		numaAffinities = append(numaAffinities, hint.NUMANodeAffinity)
+	}
+
+	// Merge the affinities using a bitwise-and operation.
+	mergedAffinity, _ := bitmask.NewBitMask(p.numaNodes...)
+	mergedAffinity.And(numaAffinities...)
+	// Build a mergedHint from the merged affinity mask.
+	return TopologyHint{mergedAffinity, preferred}
+}
+
+// Return hints that have valid bitmasks with exactly one bit set.
+func (p *singleNumaNodePolicy) filterHints(allResourcesHints [][]TopologyHint) [][]TopologyHint {
+	var filteredResourcesHints [][]TopologyHint
+	for _, oneResourceHints := range allResourcesHints {
+		var filtered []TopologyHint
+		if len(oneResourceHints) == 0 {
+			filteredResourcesHints = append(filteredResourcesHints, filtered)
+			continue
+		}
+
+		for _, hint := range oneResourceHints {
+			if hint.NUMANodeAffinity != nil && hint.NUMANodeAffinity.Count() == 1 && hint.Preferred == true {
+				filtered = append(filtered, hint)
+			}
+		}
+
+		filteredResourcesHints = append(filteredResourcesHints, filtered)
+	}
+	return filteredResourcesHints
+}
```

Comment:
Please see https://github.com/kubernetes/kubernetes/pull/85798#discussion_r359801185 for my latest comments on this function.

Reply:
Replied under this comment also.